File: AlgorithmDataStructurePythonCSharp/Python/Algorithms/binary_search.py

# Binary search works only with sorted list

def binary_search(sorted_list, lookup_item):
    low = 0
    high = len(sorted_list) - 1

    while low <= high:
        mid = (low + high)  // 2 # integer division
        guess = sorted_list[mid]

        if guess == lookup_item:
            return mid      # returns index of the item
        if guess > lookup_item:
            high = mid - 1
        else:
            low = mid + 1

    return None




# run program
sorted_list = [15, 20, 30, 45, 55, 60, 76, 84, 98, 100]
print("Item found at index: ", + binary_search(sorted_list, 60))

File: AlgorithmDataStructurePythonCSharp/CSharp/Algorithms/BinarySearch.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CSharp.Algorithms
{
    public class BinarySearch
    {
        private IList<int> _sortedList;
        private int _lookupItem;

        public BinarySearch(IList<int> sortedList, int lookupItem)
        {
            _sortedList = sortedList;
            _lookupItem = lookupItem;
        }

        public int Search()
        {
            int low = 0;
            int high = _sortedList.Count - 1;
            

            while (low <= high)
            {
                int mid = (low + high) / 2;

                if (_sortedList[mid] == _lookupItem) return mid;
                
                if (_sortedList[mid] > _lookupItem) 
                {
                    // lookup item is on the left side
                    high = mid - 1;
                }
                else // lookup item is on the right side
                {
                    low = mid + 1;
                }
            }

            return 0;
        }
    }
}
